lib/decidim/monkey_patches/mailer/notification_mailer_patches.rb

```diff
@@ -1,30 +1,35 @@
 # frozen_string_literal: true
 module NotificationMailerPatch
 
-    def send_custom_email(event, event_class_name, resource, user, extra)
-        with_user(user) do
-            @organization = resource.organization
-            event_class = event_class_name.constantize
-            @event_instance = event_class.new(resource: resource, event_name: event, user: user, extra: extra)
-            subject = @event_instance.email_subject
-    
-            mail(to: user.email, subject: subject, template_name: extra[:template])
-        end
+  def event_received(event, event_class_name, resource, user, extra) # To avoid event_received from Core to override all of our methods, we use event_received as default method then redirect to corresponding custom method when it's needed.
```

Comment:
@ludivinecp it seems we override completly the default `event_received` method that way
is that what we want?
Also then it seems we can dissmiss altogether the `email_notification_generator` patch

Reply:
Yes after reflexion it may be not the better way to do it.
Actually `event received` was overriding all my methods so I choose to override itself. But to get update from decidim I think we shouldn't do that...
So we have to find a way to get our methods working without event received interferance.
The `email_notification_generator`patch was indeed useless in this case